PostFreeSurfer/scripts/CreateMyelinMaps.sh

```diff
@@ -242,60 +241,128 @@ for STRING in "$AtlasSpaceFolder"/"$NativeFolder"@native@roi "$AtlasSpaceFolder"
   done
 done
 
-BiasFieldComputed=false
-#Reduce memory usage by smoothing on downsampled mesh (match the gifti version by using the first lowresmesh)
-LowResMesh="${LowResMeshesArray[0]}"
-# myelin map only loop
-for MyelinMap in MyelinMap SmoothedMyelinMap ; do
-	if [ "$BiasFieldComputed" = false ]; then
-		# ----- Begin moved statements -----
-		# Recompute Myelin Map Bias Field Based on Better Registration
-		log_Msg "Recompute Myelin Map Bias Field Based on Better Registration"
+# BC processing
+if [ "${T2wPresent}" = "YES" ] ; then	
+	# Create surface on HighResMesh in subject's T1w space
+	${CARET7DIR}/wb_command -surface-resample ${StudyFolder}/${Subject}/T1w/${NativeFolder}/${Subject}.L.midthickness.native.surf.gii \
+		${AtlasSpaceFolder}/${NativeFolder}/${Subject}.L.sphere.MSMSulc.native.surf.gii \
+		${AtlasSpaceFolder}/${Subject}.L.sphere.${HighResMesh}k_fs_LR.surf.gii \
+		BARYCENTRIC ${StudyFolder}/${Subject}/T1w/${Subject}.L.midthickness.${HighResMesh}k_fs_LR.surf.gii
+	${CARET7DIR}/wb_command -surface-resample ${StudyFolder}/${Subject}/T1w/${NativeFolder}/${Subject}.R.midthickness.native.surf.gii \
+		${AtlasSpaceFolder}/${NativeFolder}/${Subject}.R.sphere.MSMSulc.native.surf.gii \
+		${AtlasSpaceFolder}/${Subject}.R.sphere.${HighResMesh}k_fs_LR.surf.gii \
+		BARYCENTRIC ${StudyFolder}/${Subject}/T1w/${Subject}.R.midthickness.${HighResMesh}k_fs_LR.surf.gii
 		
-		# Myelin Map BC using low res
-		"$HCPPIPEDIR"/global/scripts/MyelinMap_BC.sh \
-			--study-folder="$StudyFolder" \
-			--subject="$Subject" \
-			--registration-name="MSMSulc" \
-			--use-ind-mean="$UseIndMean" \
-			--low-res-mesh="$LowResMesh" \
-			--myelin-target-file="$MyelinTargetFile" \
-			--map="$MyelinMap"
-		# ----- End moved statements -----
-		# bias field is computed in the module MyelinMap_BC.sh
-		BiasFieldComputed=true
-		${CARET7DIR}/wb_command -cifti-separate ${AtlasSpaceFolder}/${NativeFolder}/${Subject}.BiasField.native.dscalar.nii COLUMN \
-			-metric CORTEX_LEFT ${AtlasSpaceFolder}/${NativeFolder}/${Subject}.L.BiasField.native.func.gii \
-			-metric CORTEX_RIGHT ${AtlasSpaceFolder}/${NativeFolder}/${Subject}.R.BiasField.native.func.gii
-	else
-		# bias field in native space is already generated
-		# BC the other types of given myelin maps
-		${CARET7DIR}/wb_command -cifti-math "Var - Bias" ${AtlasSpaceFolder}/${NativeFolder}/${Subject}.${MyelinMap}_BC.native.dscalar.nii -var Var ${AtlasSpaceFolder}/${NativeFolder}/${Subject}.${MyelinMap}.native.dscalar.nii -var Bias ${AtlasSpaceFolder}/${NativeFolder}/${Subject}.BiasField.native.dscalar.nii
+	# determine the resolution of the reference myelin map
+	# TODO: is there a better way to directly get the information of 164k, 59k or 32k? 1) hack into the xml, 2) hack info file names, 3)???
```

Comment:
Probably just expect this to match HighResMesh?

Reply:
The `-cifti-separate` above are not necessary IMO. Then I am not sure if the reference myelin map is always matched with `$HighResMesh`. Otherwise, it would be nice if the script can automatically detect the resolution of the reference map and then decide whether to downsample to 32k or not.

Reply:
This is a sub-script of `PostFreeSurfer.sh`, it is very narrow-purpose (it doesn't get called by any other pipeline, or on its own, otherwise we would convert it to newopts.shlib).  If `PostFreeSurfer.sh` doesn't need that flexibility, we shouldn't add complication to try to support it.

Reply:
This is related to a mandatory argument `--refmyelinmaps` in the `PostFreeSurferPipeline.sh`, which is then passed to this sub-script. I recalled in last meeting with Matt @glasserm he wanted the script to support automatic detection of the resolution of the input reference myelin map. I would like to see if he changed his mind.

Reply:
For backwards compatibility I think we should keep the ability to use 164k files without specifying anything additional and also keep all the files that the prior command generated.